From 5bab1ad2af8ec1c85eccc9ce9de0ec0364fd7d6c Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Sat, 21 Dec 2013 20:08:06 +0000
Subject: [PATCH 13/31] Add GCC 4.8.2 patch for crtstuff.c

.. it directly included auto-host.h, it shouldn't, instead
tconfig.h includes that file, but only after defining
USED_FOR_TARGET so that things only needed for the host
are not defined when building target libraries and objects
---
 ...nclude-auto-host.h-directly-in-crtstuff.c.patch | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)
 create mode 100644 patches/gcc/4.8.2/120-dont-include-auto-host.h-directly-in-crtstuff.c.patch

diff --git a/patches/gcc/4.8.2/120-dont-include-auto-host.h-directly-in-crtstuff.c.patch b/patches/gcc/4.8.2/120-dont-include-auto-host.h-directly-in-crtstuff.c.patch
new file mode 100644
index 0000000..6b0e058
--- /dev/null
+++ b/patches/gcc/4.8.2/120-dont-include-auto-host.h-directly-in-crtstuff.c.patch
@@ -0,0 +1,22 @@
+diff -urN gcc-4.8.2.orig/libgcc/crtstuff.c gcc-4.8.2/libgcc/crtstuff.c
+--- gcc-4.8.2.orig/libgcc/crtstuff.c	2013-12-21 19:50:27.385506000 +0000
++++ gcc-4.8.2/libgcc/crtstuff.c	2013-12-21 20:02:13.816211300 +0000
+@@ -51,7 +51,17 @@
+ /* FIXME: Including auto-host is incorrect, but until we have
+    identified the set of defines that need to go into auto-target.h,
+    this will have to do.  */
+-#include "auto-host.h"
++/* Actually don't include this as it breaks building on Windows where
++   caddr_t is defined as "char *" by auto-host.h, which causes glibc's
++   sys/types.h to fall over at "typedef __caddr_t caddr_t;". auto-host.h
++   is included in-directly via tconfig.h and that correctly defines
++   USED_FOR_TARGET first meaning the define of caddr_t isn't done.
++   There is another fix for this:
++   https://sourceware.org/ml/crossgcc/2011-12/msg00077.html
++   .. but until I see this fix causing problems, it seems like the
++   cleanest way. I'm guessing the original FIXME comment has been fixed
++   by using tconfig.h and USED_FOR_TARGET?
++/* #include "auto-host.h" */
+ #undef pid_t
+ #undef rlim_t
+ #undef ssize_t
-- 
1.9.0

